app/llm_server/graph.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from functools import partial
+from typing import Literal
+
+from dotenv import load_dotenv
+from langchain.schema import HumanMessage, SystemMessage
+from langchain_huggingface import ChatHuggingFace
+from langgraph.graph import END, START, StateGraph
+from langgraph.graph.state import CompiledStateGraph
+from pydantic import BaseModel, Field
+
+from state import AgentState
+from utils.env_setup import get_device
+
+load_dotenv()
+# Bentoml 서버 로깅 설정
+bentoml_logger = logging.getLogger("bentoml")
+bentoml_logger.setLevel(logging.INFO)
+
+
+# 모델 output 형식 class
+class is_related(BaseModel):
+    """모델 판단 결과 형식 정하는 class"""
+
+    is_related: str = Field(
+        description="one of 'yes' or 'no' string value depending on the query relevance to Machine Learning, Deep Learning or Data Science"
+    )
+
+
+def check_relevance(
+    state: AgentState, model: ChatHuggingFace
+) -> Literal["format_question", END]:
+    """질문(query)가 데이터 사이언스, ML/DL 토픽과 관련이 있는지 여부를 판단합니다.
+
+    Args:
+        state: 그래프의 현재 상태가 담긴 AgenticState
+
+    Returns:
+        str: User가 입력한 query가 데이터 사이언스, ML/DL과 관련이 있는지 여부에 대한 결정
+            - "format_question": 관련 있으면 다음 node 이름 출력
+            - END: 관련 없으면 그래프 종료 (END 출력)
+    """
+    bentoml_logger.info("====check_relevance====")
+    query = state["query"]
+    model_with_tool = model.with_structured_output(is_related)
+
+    messages = [
+        SystemMessage(
+            content="""Please read the query from the user and determine whether the query is related to any of the following topics: \n 
+            - Machine Learning
+            - Deep Learning
+            - Data Science
+
+        If the query contains keyword(s) or semantic meaning related to the give topics, grade it as relevant. \n
+        Respond in JSON with `is_relavant` key.
```

Comment:
`is_related` key가 맞지 않을까요?

또는 코드 전반에 `is_related`와 `is_relevant`가 혼용되어있어서 맞추면 좋겠습니다

```suggestion
        Respond in JSON with `is_related` key.
```